File: LeetCode/Medium/1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
        int n = nums.length;
        int[] count = new int[n+1];
        ++count[0];
        int curr = 0;
        int ret = 0;

        for (int v : nums){
            if (v % 2 == 1) ++curr;
            if (curr >= k) ret += count[curr-k];
            ++count[curr];
        }

        return ret;
    }
}

File: LeetCode/Medium/1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        count = [0] * (n + 1)
        count[0] += 1
        curr = 0
        ret = 0

        for v in nums:
            if v % 2:
                curr += 1
            if curr >= k and count[curr-k]:
                ret += count[curr-k]
            count[curr] += 1
            
        
        return ret